src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -128,6 +133,126 @@ internal partial class Binder
             { WasCompilerGenerated = wasCompilerGenerated };
         }
 
+        private BoundExpression CreateImplicitNewConversion(SyntaxNode syntax, BoundExpression source, Conversion conversion, bool isCast, TypeSymbol destination, DiagnosticBag diagnostics)
+        {
+            var node = (ObjectCreationExpressionSyntax)source.Syntax;
```

Comment:
>ObjectCreationExpressionSyntax [](start = 24, length = 30)

As I mentioned in BoundNodes.xml, we should not require a relationship between an `UnboundObjectCreationExpression` and its syntax node. Please store the initializer (or anything else needed) in the `UnboundObjectCreationExpression`.